[13.0][purchase_sale_inter_company] Incoming Shipment Auto Validation

JordiBForgeFlow opened the pull request:
Before, when you completed a delivery order in company A this would automatically validate the incoming shipment in company B.
This PR introduces a parameter in the configuration 'Incoming Shipment Auto Validation' that allows you to choose if you want this behavior. Otherwise the incoming shipment needs to be manually validated.

This manual validation makes sense when the two warehouses are managed by different people, and you want to actually track the actual receipt of the goods to the destination warehouse.

cc @JoanSForgeFlow

JordiBForgeFlow committed:
[purchase_sale_inter_company] fix purchase line link with vendor bill

JordiBForgeFlow committed:
[purchase_sale_inter_company][imp] add the possibility to decide
if you want to auto-validate the incoming shipment when the delivery
order has been completed.

pedrobaeza commented:
I think the stock dependency should be extracted to an extra module, as you can have inter-company services.

JordiBForgeFlow commented:
@pedrobaeza Agreed. That would be a nice improvement, but for now it is not split. This would require a data migration. Perhaps we could target that for 15.0.

pedrobaeza commented:
I think it's not yet merged in v14, but no migration is required AFAIK.

JordiBForgeFlow commented:
It has already been merged in 14.0. The current module introduces fields in pickings, that would be moved to the new module

pedrobaeza commented:
> The current module introduces fields in pickings, that would be moved to the new module

Yes, and that's doesn't represent any migration script: just move the fields to such module.

JordiBForgeFlow commented:
Ok, will do that.

On Mon, Jul 19, 2021 at 3:58 PM Pedro M. Baeza ***@***.***>
wrote:

> The current module introduces fields in pickings, that would be moved to
> the new module
>
> Yes, and that's doesn't represent any migration script: just move the
> fields to such module.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/multi-company/pull/304#issuecomment-882570733>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB2T6VQPXMTVGAGF7NK6MTTTYQVRNANCNFSM5ATXF3HA>
> .
>


-- 
*Jordi Ballester Alomar *
CEO & Founder
Spain: (+34) 629 53 07 07 | USA: (+1) 646 980 46 59 | Denmark: (+45) 78 78
21 89
Email: ***@***.*** ***@***.***> | Twitter:
/jordibforgeflow <https://twitter.com/jordibforgeflow> | Linkedin:
/jordiballesteralomar <https://www.linkedin.com/in/jordiballesteralomar>

<https://www.forgeflow.com/>
<https://github.com/ForgeFlow>   <https://www.linkedin.com/company/2415389>
   <https://www.youtube.com/channel/UC6tv8_do-x2jg2nDx4H_DnA>
<https://twitter.com/ForgeFlow>

*The content of this email is confidential and intended for the recipient
specified in message only. It is strictly forbidden to share any part of
this message with any third party, without a written consent of the sender.
If you received this message by mistake, please reply to this message and
follow with its deletion, so that we can ensure such a mistake does not
occur in the future.*

pedrobaeza commented:
Thanks. Ping me and I'll check.

JordiBForgeFlow committed:
[account_invoice_inter_company][imp]  The extract_state field is
introduced by module 'account_invoice_extract' of Odoo Enterprise. We check
if it exists, and if it does, we set it to done so that the bill is not
sent for scanning to the external OCR.

JordiBForgeFlow committed:
[account_invoice_inter_company][fix] avoid access error

JordiBForgeFlow committed:
[purchase_sale_inter_company][fix] force company when
creating the sales order.

AdriaGForgeFlow reviewed (approved):
Code and functional review :+1:
